### ISSUE TITLE
Proposal: Executor - Scheduler interop

### ISSUE THREAD

**Issue by hzsweers** [akarnokd/RxJava2Extensions#24]
Right now RxJava has support for creating a `Scheduler` from an `Executor`, but not the other way around. This could be useful for allowing shared use of a scheduler's thread pool for APIs that don't necessarily speak RxJava.

**Comment by akarnokd** [comment]
Executor is a SAM so you can set it to Scheduler::scheduleDirect

**Comment by hzsweers** [comment]
Good point. What do you think about the more expanded ones like `ExecutorService` or `ScheduledExecutorService`? I noticed the internal `ExecutorScheduler` has some understanding of these

**Comment by akarnokd** [comment]
Do you have something specific you need an `ExecutorService` for?

**Comment by hzsweers** [comment]
Most of the use case unfortunately is just to share threadpools when bridging to non-rx code. We have a few SDKs embedded in our apps that speak `Executor` (+ friends), and since we're a mobile environment we're trying to keep threadcount down rather than provide each with their own unique thread pool

**Comment by akarnokd** [comment]
> we're a mobile environment 

You'd probably want a separate standalone library for this then as you may not need all the extensions of this library just to get the interop.

**Comment by hzsweers** [comment]
that's fair, though we're already including this library and fairly happy to rely on proguard for unused APIs. I wasn't actually sure where to ask, and also maybe looking for a bit of guidance on implementing such a thing if you have any :)

**Comment by akarnokd** [comment]
You have to implement interfaces and delegate to the relevant `schedule` calls. You will likely need some `CountDownLatch`es for the `invokeAll`/`invokeAny`.
